app/src/main/java/com/bootcamp/kotlin/search/SearchFragment.kt

```diff
@@ -0,0 +1,72 @@
+package com.bootcamp.kotlin.search
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.bootcamp.kotlin.databinding.FragmentSearchBinding
+import com.bootcamp.kotlin.movies.Movie
+import com.bootcamp.kotlin.movies.MoviesRepositoryImpl
+import com.bootcamp.kotlin.movies.adapter.MoviesAdapter
+import kotlinx.android.synthetic.main.fragment_search.*
+import kotlinx.android.synthetic.main.view_progress_bar.*
+
+class SearchFragment : Fragment(), SearchContract.View{
+
+    private lateinit var binding:FragmentSearchBinding
+    private var listener:Listener? = null
+    private var presenter:SearchContract.Presenter?=null
+
+   companion object{
+       @JvmStatic
+       fun newInstance():SearchFragment = SearchFragment()
+   }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentSearchBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        presenter = SearchPresenter(view = this, repository = MoviesRepositoryImpl())
+        presenter?.initView()
+        search_btn.setOnClickListener{
+            presenter?.searchMovies(search_src_text.text.toString())
+        }
+    }
+
+    override fun showMovies(movies: List<Movie>) {
+        binding.moviesRecyclerView.adapter = adapter
+        adapter.movies = movies
+    }
+
+    override fun showProgress(isVisible: Boolean) {
+        progress.visibility = if (isVisible) View.VISIBLE else View.GONE
+    }
+
+    interface Listener {
```

Comment:
Haciendo referencia al (activity as MainActivity).navigateTo(it), podrías entrar al Detalle de la pelicula

Reply:
no se que tan buena idea se acceder de esta forma ya que con herencia se vuelven dependientes y si en algun momento decides de que  ya no sera el main activity o eliminas el mainactivity y creas el initactivity vas a tener que ir clase por clase a cambiarlo. ademas si este fragment lo mueves a otra actividad porque ya no ira en el main activity tendras que modificar ese codigo, con lo simple que solo seria implementar la interface en la actividad que deses